wrapper/wrapper

```diff
@@ -1,10 +1,28 @@
 #!/bin/bash
 set -e
 
-if [ $(id -u) == 0 ] || [ -v DISPLAY ] || snapctl is-connected login-session-control; then
-  exec desktop-launch $@
-else
+if [ $(id -u) != 0 ] && ! [ -v DISPLAY ] && ! snapctl is-connected login-session-control; then
   echo "ERROR: You need one of: root, a DISPLAY, or to connect the login-session-control interface."
   echo "e.g. sudo snap connect" $SNAP_NAME":login-session-control :login-session-control"
   exit 1
 fi
+
+if [ -v MIR_SERVER_WAYLAND_HOST ]; then
+  MIR_SERVER_CONSOLE_PROVIDER=${MIR_SERVER_CONSOLE_PROVIDER:-none}
+else
+  if [ $(id -u) != 0 ]; then
+    MIR_SERVER_CONSOLE_PROVIDER=${MIR_SERVER_CONSOLE_PROVIDER:-$(snapctl get console-provider)}
+  else
+    MIR_SERVER_CONSOLE_PROVIDER=${MIR_SERVER_CONSOLE_PROVIDER:-auto}
+  fi
+fi
```

Comment:
Why would we not respect what's in `snapctl get console-provider` across the board if it's not overridden by environment?

Reply:
1. That's the existing behaviour.

2. It is also convenient: when running as a daemon (and root) we need to use `console-provider=vt`, but if running as a normal user in a desktop session we typically want `auto`.

Reply:
Well, yes. But that's why the environment can override (which we didn't have before)… If `snap set…` is only applied to the daemon (or to root user, rather), shouldn't those options be prefixed with `root_` or some such?

Reply:
As things are before these changes, you can `snap install mir-test-tools` on the desktop or a VT and run the tests:
```plain
mir-test-tools.smoke-test
```
With the changes as proposed that remains true,

With your suggestion that becomes:
```plain
MIR_SERVER_CONSOLE_PROVIDER=auto mir-test-tools.smoke-test
```
I don't see why you view that as an improvement.

Reply:
Or, should we not have a `snapctl` setting for `console-provider` at all?

Reply:
That may well be DTRT. If a `snap set` doesn't affect (or shouldn't) "the whole" of the snap, maybe it shouldn't exist.